bpf/bpf_sock.c

```diff
@@ -275,6 +275,7 @@ static __always_inline int __sock4_xlate_fwd(struct bpf_sock_addr *ctx,
 	struct lb4_key key = {
 		.address	= ctx->user_ip4,
 		.dport		= ctx_dst_port(ctx),
+		.proto		= (__u8)ctx->protocol,
```

Comment:
small nit: the __u8 cast should not be needed here and in the others. Also I'd just fetch `__u32 proto = ctx->protocol` just once and then use it also for e.g. `sock_proto_enabled ()` below etc so compiler can optimize here. Two fetches on ctx directly it cannot optimize.